lib/riemann/helpers/event.ex

```diff
@@ -59,7 +40,46 @@ defmodule Riemann.Helpers.Event do
         |> Map.delete(:metric_sint64)
         |> Map.put(:attributes, attributes)
       end
+    end
+  end
+
+  @moduledoc false
+  def build(args, mod) do
+    args
+    |> Enum.into(%{})
+    |> Map.put_new(:time, :erlang.system_time(:seconds))
+    |> Map.put_new_lazy(:host, &default_event_host/0)
+    |> set_attributes_field
+    |> set_metric_pb_fields
+    |> Map.to_list
+    |> mod.new()
+  end
 
+  defp default_event_host do
+    event_host_setting() || machine_hostname()
+  end
+
+  defp set_attributes_field(map) do
```

Comment:
Hm, how about instead of modifying `build/2`, just this?

```elixir
defp set_attributes_field(%{attributes: a} = map) when not is_nil(a) do
  Map.put(map, :attributes, Attribute.build(a))
end
# catches both nil and missing cases
defp set_attributes_field(map), do: map
```